dialog/custom_test.go

```diff
@@ -142,3 +142,33 @@ func TestCustomConfirm_Importance(t *testing.T) {
 	d.Show()
 	test.AssertRendersToImage(t, "dialog-custom-confirm-importance.png", w.Canvas())
 }
+
+func TestCustomSetIconSuccess(t *testing.T) {
+	test.NewTempApp(t)
+	w := test.NewTempWindow(t, canvas.NewRectangle(color.Transparent))
+	size := fyne.NewSize(200, 300)
+	w.Resize(size)
+
+	test.ApplyTheme(t, test.Theme())
+	label := widget.NewLabel("Test was successful.")
+	d := NewCustom("Test result", "Close", label, w)
+	d.SetIcon(theme.ConfirmIcon())
+	d.Show()
+
+	test.AssertRendersToImage(t, "dialog-custom-seticon-success.png", w.Canvas())
+}
+
+func TestCustomSetIconNil(t *testing.T) {
+	test.NewTempApp(t)
+	w := test.NewTempWindow(t, canvas.NewRectangle(color.Transparent))
+	size := fyne.NewSize(200, 300)
+	w.Resize(size)
+
+	test.ApplyTheme(t, test.Theme())
+	label := widget.NewLabel("Test was successful.")
+	d := NewCustom("Fyne test", "Dimiss", label, w)
+	d.SetIcon(nil)
+	d.Show()
+
+	test.AssertRendersToImage(t, "dialog-custom-seticon-nil.png", w.Canvas())
```

Comment:
You don't need two different test functions just for setting different values. There is just unnecessary code duplication between them. I'd rather call the test `TestCustom_SetIcon` to be consistent with the other tests (that is the standard for naming) and then just at the end of the first one you have, set the icon to nil and do a new assert to image.